File: proveedor-service/src/main/java/com/MilkStgo/proveedorservice/services/ProveedorService.java
package com.MilkStgo.proveedorservice.services;

import com.MilkStgo.proveedorservice.entities.ProveedorEntity;
import com.MilkStgo.proveedorservice.repositories.ProveedorRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.ArrayList;

@Service
public class ProveedorService {
    @Autowired
    ProveedorRepository proveedorRepository;

    public String obtenerCategoria(String codigo){
        return proveedorRepository.findByCodigo(codigo).getCategoria();
    }

    public String obtenerNombre(String codigo){ return proveedorRepository.findByCodigo(codigo) != null? proveedorRepository.findByCodigo(codigo).getNombre():"no name";}

    public String obtenerRetencion(String codigo){ return proveedorRepository.findByCodigo(codigo).getRetencion();}

    public void guardarProveedor(String codigo, String nombre, String categoria, String retencion){
        ProveedorEntity proveedor = new ProveedorEntity();
        proveedor.setCodigo(codigo);
        proveedor.setNombre(nombre);
        proveedor.setCategoria(categoria);
        proveedor.setRetencion(retencion);
        proveedorRepository.save(proveedor);
    }

    public ArrayList<ProveedorEntity> obtenerProveedores(){
        return (ArrayList<ProveedorEntity>) proveedorRepository.findAll();
    }
}

File: frontend/src/services/ProveedorService.ts
import axios from 'axios';

const url = 'http://localhost:8080/proveedor';

class ProveedoresService {
  obtenerProveedores() {
    return axios.get(url);
  }

  nuevoProveedor(codigo: string, nombre: string, categoria: string, retencion: string) {
    const proveedor = {
      codigo: codigo,
      nombre: nombre,
      categoria: categoria,
      retencion: retencion
    };
    return axios.post(url, proveedor);
  }
}

const instance = new ProveedoresService();
export default instance;